documentation/book/con-certificate-renewal.adoc

```diff
@@ -0,0 +1,69 @@
+
+When a CA certificate expires, the certificates signed by it will fail validation, even if they are otherwise valid.
+This means when a CA certificate is replaced it is also necessary to replace all the certificates previously signed by that CA.
+
+While CA certificates are being replaced it is necessary for peers to trust certificates signed by either the old or the new CA. 
+This ensures continued operation of the cluster.
+
+In order to allow the CA certificates to be renewed without loss of service, Cluster Operator will initiate certificate renewal before the CA certificates expire.
+This can be configured in `Kafka.spec.clusterCa.renewalDays`, and `Kafka.spec.clientsCa.renewalDays`.
+Both default to 30 days.
+This renewal period is measured (backwards in time) from the date that the current certificate will expire.
+
+[source]
+----
+Not Before                                     Not After
+    |                                              |
+    |<--------------- validityDays --------------->|
+                              <--- renewalDays --->|
+----
+
+What the Cluster Operator does during this renewal period depends whether the relevant `Kafka.spec.clusterCa.generateCertificateAuthority` or `Kafka.spec.clientsCa.generateCertificateAuthority` is enabled.
+
+== Renewal process with generated CAs
+
+The Cluster Operator used the following process CA certificate renewal:
+
+. Generate a new CA key and certificate. The private key replaces the old private key in the corresponding `Secret`. The new certificate is given the name `ca.crt` within the corresponding `Secret` and the old certificate is renamed `ca-_<expiry-date>_.crt`.
+
+. Restart Zookeeper nodes so that they will trust the new CA certificate
+
+. Restart Kafka brokers so that they will trust the new CA certificate
+
+. Restart the Topic and User operators so that they will trust the new CA certificate
+
+. Generate new client certificates (for Zookeeper nodes, Kafka brokers, and the entity operator) signed by the new CA.
+
+. Perform the same restarts so that clients are using certificates signed by the new CA certificate.
+
+At the end of the renewal period the Cluster Operator will remove the now expired CA certificates (those named `ca-_<expiry-date>_.crt`) from the corresponding `Secret` and perform another round of restarts.
+
+== Renewal process with your own CA certificates
+
+At the start of the renewal period the Cluster Operator will start logging at the `WARN` level that new CA certificates and keys are needed. Ones these have been provided the Cluster Operator will perform the same sequence of restarts within the cluster is it would for renewal with generated CAs.
+
+== Client applications
+
+The Cluster Operator does know about all the client applications using the Kafka cluster.
+
+IMPORTANT: Depending on how your applications are configured you may need take action to ensure they continue working.
+
+The important points are:
```

Comment:
About the 3rd bullet: The point I'm trying to make is that generating new certificates and keys must be done _during_ the renewal period, because if you haven't done it by the end of the period your clients won't be able to connect any more. The suggested rewording loses this. This is why the renewal period is a configurable thing: So users can configure at long as they need to do the manual things in the 2nd and 3rd bullets.

How about:

```
* When renewing client certificates, if you are provisioning 
client certificates and keys manually, you 
must generate new client certificates and ensure the 
new certificates are used by clients within the 
renewal period. 
Failure to do this by the end of the renewal period 
could result in client applications being unable to 
connect.
```

Reply:
That makes sense - please use your text for the third bullet.